src/NinjaTools.FluentMockServer.API/Proxy/Visitors/Collections/HeaderCollection.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using NinjaTools.FluentMockServer.API.Tests.Serializtion.Wrapping;
+using NinjaTools.FluentMockServer.API.Types;
+
+namespace NinjaTools.FluentMockServer.API.Proxy.Visitors.Collections
+{
+
+    [JsonConverter(typeof(UnwrapConverter))]
+    public class HeaderCollection : IVisitable,  IDontRenderWhenEmpty, IUnwrappable
+    {
+        public HeaderCollection()
+        { }
+
+        /// <inheritdoc cref="HeaderCollection()" />
+        public HeaderCollection(IDictionary<string, string[]> dictionary) : this()
+        {
+            dictionary ??= new Dictionary<string, string[]>();
+            foreach (var (k,v) in dictionary)
+                Headers.Add(k, v);
+        }
+
+        /// <inheritdoc cref="HeaderCollection()" />
+        public HeaderCollection(params (string, string[])[] kvps) : this()
+        {
+            kvps ??= new (string, string[])[0];
+            foreach (var (k,v) in kvps)
+                Headers.Add(k, v);
+        }
+
+        public Dictionary<string, string[]> Headers { get; set; } =  new Dictionary<string, string[]>();
+
+        /// <inheritdoc />
+        public bool IsEnabled() => Headers.Any();
+        /// <inheritdoc />
+        public object Unwrap() => Headers;
+
+        /// <inheritdoc />
+        public void Accept(IVisitor visitor)
+        {
+            if (visitor is IVisitor<HeaderCollection> typed)
+            {
+                typed.Visit(this);
+            }
+        }
+    }
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+//
+// //    [JsonConverter(typeof(HeaderCollectionConverter))]
+//     public class HeaderCollection : Dictionary<string, string[]>, IVisitable,  IDontRenderWhenEmpty
+//     {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)